include/envoy/upstream/load_balancer.h

```diff
@@ -43,6 +45,33 @@ class LoadBalancerContext {
    * balancing.
    */
   virtual const Http::HeaderMap* downstreamHeaders() const PURE;
+
+  /**
+   * Called to retrieve a reference to the priority load data that should be used when selecting a
+   * priority. Implementations may return the provided original reference to make no changes, or
+   * return a reference to alternative PriorityLoad held internally.
+   *
+   * @param priority_state current priority state of the cluster being being load balanced.
+   * @param original_priority_load the cached priority load for the cluster being load balanced.
+   * @return a reference to the priority load data that should be used to select a priority.
+   *
+   */
+  virtual const PriorityLoad&
+  determinePriorityLoad(const PrioritySet& priority_set,
+                        const PriorityLoad& original_priority_load) PURE;
+
+  /**
+   * Called to determine whether we should reperform host selection. The load balancer
+   * will retry host selection until either this function returns true or hostSelectionRetryCount is
+   * reached.
+   */
+  virtual bool shouldSelectAnotherHost(const Host& host) PURE;
+
+  /**
+   * Called to determine how many times host selection should be retried until the filter is
+   * ignored.
+   */
+  virtual uint32_t hostSelectionRetryCount() PURE;
```

Comment:
Should this be `hostSelectionRetryCount() const`? I don't see any scenario where this might mutate state in an implementation.

Reply:
Yeah I think this should be `const`